#21 [BE] feat: 마일스톤 추가 기능 구현

AlbertImKr opened the pull request:
## 🔑 Key changes
- [x] 마일스톤 추가 기능 구현

## 👋 To reviewers
- OrganizationRepository 구현

AlbertImKr committed:
feat: 마일스톤 추가 기능 구현

CDBchan commented on be/src/test/java/com/codesquad/issuetracker/api/milestone/acceptance/MilestoneAcceptanceTest.java:
```diff
@@ -0,0 +1,27 @@
+    @Test
+    void createMilestone() {
+        // given
+        var response = 마일스톤을_생성한다();
```
이런 방식이 ATDD 방식인건가요? 처음보는 형태라 신기하네요.

kimm-charlie replied:
이런 방식이 ATDD 방식인건가요? 처음보는 형태라 신기하네요.

kimm-charlie commented on be/src/main/java/com/codesquad/issuetracker/api/milestone/repository/MilestoneRepositoryImpl.java:
```diff
@@ -0,0 +1,36 @@
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+        jdbcTemplate.update(saveMilestone, sqlParameterSource, keyHolder);
+        Number key = keyHolder.getKey();
+        return key != null ? key.longValue() : null;
```
이부분은 null 값을 직접적으로 다루는 대신 optional 을 사용해서 다뤄도 좋을거 같습니다. 
```
Optional.ofNullable(key).map(Number::longValue);
```

AlbertImKr replied:
좋은 방법인것 같아요. 수정할게요.

kimm-charlie commented on be/src/main/java/com/codesquad/issuetracker/api/milestone/service/MilestoneService.java:
```diff
@@ -0,0 +1,41 @@
+    public long create(String organizationTitle, MilestoneCreatRequest mileStoneCreatRequest) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_TIME_PATTERN);
+        LocalDate localDate = LocalDate.parse(mileStoneCreatRequest.getDueDate(), formatter);
+        Long organizationId = organizationRepository.findIdByTitle(organizationTitle);
```
organizationRepository에서 title을 사용하여  id를 가져오는 메서드명은 추후 be멤버끼리 상의한후 정해야할거 같습니다.
label 생성때도 해당 메서드가 필요하여 아래와 같이 정해놓은 상태인데 같이 의논후 결정하면 좋을거 같습니다.
```
 Long findBy(String organizationTitle);
```

AlbertImKr replied:
좋아요!

kimm-charlie commented on be/src/main/java/com/codesquad/issuetracker/api/milestone/service/MilestoneService.java:
```diff
@@ -0,0 +1,41 @@
+    private final MilestoneRepository milestoneRepository;
+    private final OrganizationRepository organizationRepository;
+
+    public long create(String organizationTitle, MilestoneCreatRequest mileStoneCreatRequest) {
```
milestoneCreateRequest 오타 수정해야할것 같습니다.

AlbertImKr replied:
수정할게요.

kimm-charlie commented on be/src/main/java/com/codesquad/issuetracker/api/milestone/service/MilestoneService.java:
```diff
@@ -0,0 +1,41 @@
+        if (organizationId == null) {
+            throw new RuntimeException("organization이 존재하지 않습니다.");
+        }
+        Milestone milestone = Milestone.builder()
+                .organizationId(organizationId)
+                .title(mileStoneCreatRequest.getTitle())
+                .description(mileStoneCreatRequest.getDescription())
+                .isClosed(false)
+                .dueDate(localDate)
+                .build();
```
dto 안에서 toEntity() 등의 정적 팩토리 메서드를 사용해서 변환해도 좋을것 같습니다.

kimm-charlie reviewed (approved):
코드 잘봤습니다!

kimm-charlie committed:
#20 [BE] feat: 레이블 CRUD 기능 구현  (#27)

* chore: 테스트 환경 yml 파일 분리

* feat: 테스트를 위한 커스텀 어노테이션 구현

* feat: Label Entity 구현

* feat: Label DTO 구현

* feat: Organization Repository 구현

* feat: Label Repository 구현

* feat: Label Service 구현

* feat: Label Controller 구현

* refactor: 레이블 리팩토링

팀원끼리 정한 코딩 컨벤션대로 코드를 수정했다.

* docs: 환경별 DB설정파일 수정 및 스키마 추가

* refactor: 리뷰반영 및 코드 reformat

queryForObject대신 query를 사용하도록 변경
및 코드를 google style로 reformat

---------

Co-authored-by: jinny-l <jinny.l.dev@gmail.com>

AlbertImKr committed:
feat: 마일스톤 추가 기능 구현

AlbertImKr committed:
Merge branch '21-be-마일스톤-추가-기능' of https://github.com/eojjeogo-jeojjeogo/issue-tracker-max into 21-be-마일스톤-추가-기능

# Conflicts:
#	be/src/main/java/com/codesquad/issuetracker/api/milestone/controller/MilestoneController.java
#	be/src/main/java/com/codesquad/issuetracker/api/milestone/repository/MilestoneRepository.java
#	be/src/main/java/com/codesquad/issuetracker/api/milestone/repository/MilestoneRepositoryImpl.java
#	be/src/main/java/com/codesquad/issuetracker/api/milestone/service/MilestoneService.java
#	be/src/main/java/com/codesquad/issuetracker/api/organization/repository/OrganizationRepository.java
#	be/src/main/java/com/codesquad/issuetracker/api/organization/repository/OrganizationRepositoryImpl.java
#	be/src/test/java/com/codesquad/issuetracker/api/milestone/controller/MilestoneControllerTest.java

AlbertImKr committed:
merge: Conflict 해결
